mobilecoind-dev-faucet/src/worker.rs

```diff
@@ -529,6 +533,12 @@ impl WorkerTokenState {
             }
         });
 
+        // Update the status of in-flight txs.
+        // Doing this before checking for new UTXOs avoids some races, where we find out
+        // that our TX settled but still think that one of the UTXOs that it used is
+        // unspent.
+        self.check_on_in_flight_txs(client, logger);
```

Comment:
Moving this line here before the GetUnspentTxOutListRequest, instead of after, is the only functional change of this PR

Reply:
I think that the associated comment on check_on_in_flight_txs should also be updated to reflect this move. I'm sorry to have to ask for additional clarification but I'm a little confused about how this reduces the race condition. If the tx settles after getunspenttxoutlist is called, doesn't that still cause problems?

Reply:
The race condition that I think I see in testing is:

0. Worker has a tx is inflight, spending a TXO, lets call it A.
1. worker asks mobilecoind for its utxos, and this list  includes A.
2. worker asks mobilecoind what txs are in flight, and finds that the Tx is now settled and A is not unspent anymore. But, A is still in the list it got before. This causes it to be selected for future Tx's later.

If we do it the other way, it's possible that we think a Tx is still in flight, but one of the UTXOs it spent isn't on the list anymore. However, this doesn't cause the same problems, because we don't end up selecting a UTXO that isn't valid anymore. I hope that makes sense. I'm not sure what more comments I can make that would be appropriate without being TMI.

Reply:
That does make sense. I think that the example helped a lot. Thank you.
My other comment was just that the function check_on_in_flight_txs says that it's called before step_3/4, which is no longer accurate.

Reply:
okay thank you!